core/src/main/java/dk/alexandra/fresco/framework/configuration/NetworkConfigurationImpl.java

```diff
@@ -41,5 +45,17 @@ public String toString() {
         + parties + "]";
   }
 
+  /**
+   * Verifies that all party addresses are unique.
+   */
+  private static void checkAddressesUnique(Map<Integer, Party> parties) {
+    Set<String> addresses = new HashSet<>();
+    for (Party party : parties.values()) {
+      addresses.add(party.getHostname() + " " + party.getPort());
+    }
+    if (addresses.size() != parties.size()) {
+      throw new IllegalArgumentException("Party addresses must be unique: " + addresses);
```

Comment:
Is it useful to include `addresses` here? I would think `parties` would be more help because you can then inspect which parties had colliding addresses.